File: Sorting/Quick Sort/QuickSort.cpp
#include<iostream>
using namespace std;

int Partition(int arr[], int start, int end)
{
    int pIndex=start, pivot=arr[end];
    for(int i=start; i<end; i++)
    {
        if(arr[i]<=pivot)
        {
            int temp=arr[i];
            arr[i]=arr[pIndex];
            arr[pIndex]=temp;
            pIndex+=1;
        }
    }
    int temp=arr[end];
    arr[end]=arr[pIndex];
    arr[pIndex]=temp;
    return pIndex;
}
void quicksort(int arr[], int start, int end)
{
    if(start<end)
    {
        int part=Partition(arr, start, end);
        quicksort(arr, start, part-1);
        quicksort(arr, part+1, end);
    }
}
int main()
{
    int arr[100], n;
    cout << "Enter size: " << endl;
    cin >> n;
    cout << "Enter array: " << endl;
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    quicksort(arr, 0, n - 1);
    cout << "Sorted array..." << endl;
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}

File: Sorting/Quick Sort/QuickSort.java
import java.util.Scanner;

public class QuickSort {
    
    static Scanner scn = new Scanner(System.in);

    public static void display(int[] arr) {
        for(int i = 0 ; i < arr.length ; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void input(int[] arr) {
        for(int i = 0; i < arr.length; i++) {
            arr[i] = scn.nextInt();
        }
    }

    public static int parition(int[] arr , int low, int high) {
        int i = low , j = low, pivot = arr[high];
        while(j <= high ){
            if(arr[j] <= pivot){
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                i++;
            }
            j++;
        }
        return i - 1;
    }

    public static void quickSort(int[] arr, int low, int high) {
        if( low >= high ){
            return;
        }
        int pivotIndex = parition(arr, low, high);
        quickSort(arr, low ,pivotIndex - 1);
        quickSort(arr, pivotIndex + 1, high);
    }


    public static void main(String[] args) {
        System.out.println("Enter the number of elements");
        int n = scn.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements of the array");
        input(arr);
        quickSort(arr, 0, arr.length - 1);
        display(arr);
    }
}